desktop/core/src/desktop/lib/botserver/views_tests.py

```diff
@@ -47,4 +52,56 @@ def test_say_hi_user(self):
       }
       response = say_hi_user("channel", "user_id")
       api_call.assert_called_with(api_method='chat.postMessage', json={'channel': 'channel', 'text': 'Hi <@user_id> :wave:'})
-      assert_true(response['ok'])
\ No newline at end of file
+      assert_true(response['ok'])
+  
+  def test_handle_on_message(self):
+    with patch('desktop.lib.botserver.views.say_hi_user') as say_hi_user:
+      
+      response = handle_on_message("channel", "bot_id", "text", "user_id")
+      assert_equal(response.status_code, 200)
+      assert_false(say_hi_user.called)
+      
+      handle_on_message("channel", None, None, "user_id")
+      assert_false(say_hi_user.called)
+
+      handle_on_message("channel", None, "text", "user_id")
+      assert_false(say_hi_user.called)
+
+      handle_on_message("channel", None, "hello hue test", "user_id")
+      assert_true(say_hi_user.called)
+
+  def test_handle_on_link_shared(self):
+    with patch('desktop.lib.botserver.views.slack_client.chat_unfurl') as chat_unfurl:
+      with patch('desktop.lib.botserver.views._make_unfurl_payload') as mock_unfurl_payload:
+
+        client = make_logged_in_client(username="test", groupname="default", recreate=True, is_superuser=False)
+        user = User.objects.get(username="test")
+        channel_id = "channel_id"
+        message_ts = "12345.123"
+
+        # qhistory link
+        links = [{"url": "https://demo.gethue.com/hue/editor?editor=123456"}]
+        doc_data = {
+          "dialect": "mysql",
+          "snippets": [{
+            "database": "hue",
+            "statement_raw": "SELECT 5000",
+          }]
+        }
+
+        Document2.objects.create(id=123456, data=json.dumps(doc_data), owner=user)
```

Comment:
This is good to ship for now, but let's see if this one can be mocked too, e.g. one way:

with patch('desktop.lib.botserver.views.Document2.objects.get') as document2_objects_get:
  document2_objects_get.return_value = Mock(data=doc_data)

Reply:
And also those docs need to be deleted:

query = Document2.objects.create(id=123456, data=json.dumps(doc_data), owner=user)
try:
  handle.....
finally:
  query.delete()

If not sooner or later tests won't be re-entrant/idempotent

Reply:
Added the try/finally in the UT @romainr

Reply:
> This is good to ship for now, but let's see if this one can be mocked too, e.g. one way:
> 
> ```
> with patch('desktop.lib.botserver.views.Document2.objects.get') as document2_objects_get:
>   document2_objects_get.return_value = Mock(data=doc_data)
> ```

Although I've added try/finally for the time being but when I tried the above suggestion, encountered this error.
Any comments on how to go resolve it? @romainr 

<img width="1178" alt="Screenshot 2021-02-27 at 11 00 15 PM" src="https://user-images.githubusercontent.com/42064744/109394921-8f63ed00-794f-11eb-8385-cc1440a6be0b.png">

Reply:
Fixed the error and updated the UT by mocked `Document2.objects.get` and `_get_gist_document` @romainr